From b2a7d58773959d82a0ae00b792479de3fa049cb4 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 14 Oct 2021 17:17:36 -0700
Subject: [PATCH] builder: Add a bundle's packages as the content to the
 tracking file

Add the packages found in a bundle to the bundle tracking file. This
will be handy when trying to figure out client side what packages are
installed on the system.

Signed-off-by: William Douglas <william.douglas@intel.com>
---
 builder/bundles.go | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/builder/bundles.go b/builder/bundles.go
index 4b18db9..fea0ef2 100644
--- a/builder/bundles.go
+++ b/builder/bundles.go
@@ -837,6 +837,22 @@ func buildFullChroot(b *Builder, set *bundleSet, packagerCmd []string, buildVers
 	return installSpecialFilesToFull(b, packagerCmd, set, fullDir, version)
 }
 
+func getBundlePkgNames(b *bundle) []string {
+	keys := make([]string, len(b.AllRpms))
+	i := 0
+	for _, pkgInfo := range b.AllRpms {
+		keys[i] = pkgInfo.name
+		i++
+	}
+	return keys
+}
+
+func getBundlePkgNamesSorted(b *bundle) []string {
+	keys := getBundlePkgNames(b)
+	sort.Strings(keys)
+	return keys
+}
+
 // installSpecialFilesToFull installs the bundle tracking files and installs special case files for the
 // os-core and update bundles.
 func installSpecialFilesToFull(b *Builder, packagerCmd []string, set *bundleSet, fullDir, version string) error {
@@ -846,7 +862,8 @@ func installSpecialFilesToFull(b *Builder, packagerCmd []string, set *bundleSet,
 		return err
 	}
 	for _, bundle := range *set {
-		if err := ioutil.WriteFile(filepath.Join(bundleDir, bundle.Name), nil, 0644); err != nil {
+		data := []byte(strings.Join(getBundlePkgNamesSorted(bundle), "\n") + "\n")
+		if err := ioutil.WriteFile(filepath.Join(bundleDir, bundle.Name), data, 0644); err != nil {
 			return err
 		}
 	}
-- 
2.33.0

